enable test_sampled_addmm_zero_sized_cuda for rocm

dnikolaev-amd opened the pull request:
Enable test_sampled_addmm_zero_sized_cuda_* only for ROCm and CUDA issue is currently active.
Passed since ROCm 5.6

test_sampled_addmm_zero_sized_cuda_float32
test_sampled_addmm_zero_sized_cuda_float64
test_sampled_addmm_zero_sized_cuda_complex64
test_sampled_addmm_zero_sized_cuda_complex128

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang

dnikolaev-amd committed:
enable test_sampled_addmm_zero_sized_cuda for rocm

dnikolaev-amd committed:
remove unused skipCUDAIfRocm from import

pruthvistony commented on test/test_sparse_csr.py:
```diff
@@ -2469,9 +2469,8 @@ def test_sampled_addmm_autograd(self, device, dtype):
-    @skipCUDAIfRocm
     @onlyCUDA
-    @skipCUDAIf(True, "Causes CUDA memory exception, see https://github.com/pytorch/pytorch/issues/72177")
+    @skipCUDAIf(not TEST_WITH_ROCM, "Causes CUDA memory exception, see https://github.com/pytorch/pytorch/issues/72177")
```
Please update the comment to indicate - it works on ROCm and CUDA issue is currently active.

dnikolaev-amd committed:
add comment line 

mentioned in the comment that test works on ROCm and CUDA issue is currently active

dnikolaev-amd commented:
@pytorchbot rebase

dnikolaev-amd commented:
Enabled tests
```xml
<testcase classname="TestSparseCSRCUDA" name="test_sampled_addmm_zero_sized_cuda_complex128" time="0.002" file="test_sparse_csr.py"/>
<testcase classname="TestSparseCSRCUDA" name="test_sampled_addmm_zero_sized_cuda_complex64" time="0.002" file="test_sparse_csr.py"/>
<testcase classname="TestSparseCSRCUDA" name="test_sampled_addmm_zero_sized_cuda_float32" time="0.002" file="test_sparse_csr.py"/>
<testcase classname="TestSparseCSRCUDA" name="test_sampled_addmm_zero_sized_cuda_float64" time="0.002" file="test_sparse_csr.py"/>
```

dnikolaev-amd commented:
@pytorchbot rebase

dnikolaev-amd commented:
@pytorchbot rebase

pruthvistony commented:
@malfet 
Can you please review this PR.

dnikolaev-amd commented:
@pytorchbot rebase

dnikolaev-amd commented:
@pytorchbot rebase

dnikolaev-amd commented:
@pytorchbot rebase

jithunnair-amd commented:
@malfet Please review. Few lines of change to enable some UTs

malfet commented:
@pytorchbot merge -f "It affects only ROCM and I assume it had green signal before"
